### PR TITLE
Add IntOrUUIDConverter URL converter

### PR DESCRIPTION
**Pull request by fao89**
Ref: https://docs.djangoproject.com/en/4.2/topics/http/urls/#registering-custom-path-converters

## Description
This PR adds a new `IntOrUUIDConverter` URL converter that can handle both integer and UUID values in Django URL patterns. This converter provides a flexible way to accept either integers or UUIDs in a single URL parameter, which is useful for APIs that need to support both legacy integer IDs and newer UUID-based identifiers.

- **What is being changed?** Adding a new URL converter class `IntOrUUIDConverter` in `ansible_base.lib.utils.converters`
- **Why is this change needed?** To support URL patterns that can accept either integer or UUID parameters
- **How does this change address the issue?** The converter uses regex matching and type conversion to handle both data types seamlessly

## Type of Change
- [x] New feature (non-breaking change which adds functionality)

## Self-Review Checklist
- [x] I have performed a self-review of my code
- [x] I have added relevant comments to complex code sections
- [x] I have updated documentation where needed
- [x] I have considered the security impact of these changes
- [x] I have considered performance implications
- [x] I have thought about error handling and edge cases
- [x] I have tested the changes in my local environment

### Prerequisites
- Django development environment set up
- pytest installed

### Steps to Test
1. Import the converter: `from ansible_base.lib.utils.converters import IntOrUUIDConverter`
2. Test integer conversion: `converter.to_python("123")` should return `int(123)`
3. Test UUID conversion: `converter.to_python("550e8400-e29b-41d4-a716-446655440000")` should return a `UUID` object
4. Test invalid input: `converter.to_python("invalid")` should raise `ValueError`
5. Run the test suite: `pytest test_app/tests/lib/utils/test_converters.py`

### Expected Results
- Integer strings are converted to Python `int` objects
- UUID strings are converted to Python `UUID` objects  
- Invalid inputs raise `ValueError` with descriptive message
- The `to_url()` method converts both types back to string representation
- All tests in the comprehensive test suite pass

### Implementation Details
- Uses a regex pattern that matches both integers (`[0-9]+`) and standard UUID format
- Prioritizes integer conversion (checks `isdigit()` first) for performance
- Handles edge cases like uppercase UUIDs, very large integers, and invalid formats
- Includes comprehensive test coverage (169 lines of tests for 23 lines of code)

### Required Actions
- [ ] Requires documentation updates
- [ ] Requires downstream repository changes
- [ ] Requires infrastructure/deployment changes
- [ ] Requires coordination with other teams
- [ ] Blocked by PR/MR: #XXX

### CODE REVIEW

**Review by thenets — APPROVED** [review]

File: ansible_base/lib/utils/converters.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+Reference:
+- https://docs.djangoproject.com/en/4.2/topics/http/urls/#registering-custom-path-converters
+- https://github.com/django/django/blob/fda3c1712a1eb7b20dfc91e6c9abae32bd64d081/django/urls/converters.py
+"""
+
+import uuid
+
+
+class IntOrUUIDConverter:
+    regex = "([0-9]+|[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})"
```
Comment (zkayyali812) [inline ansible_base/lib/utils/converters.py:11]: Is this regex actually used anywhere? Or can this be removed?

Comment (zkayyali812) [reply ansible_base/lib/utils/converters.py:11]: nvm, I see now, according to reference docs

**Review by zkayyali812 — APPROVED** [review]

File: ansible_base/lib/utils/converters.py (hunk 1 above)
Comment (fao89) [reply ansible_base/lib/utils/converters.py:11]: you were faster than me 😅
